crates/worker/src/partition_processor_manager/processor_state.rs

```diff
@@ -350,28 +352,23 @@ impl ProcessorState {
         }
     }
 
-    pub fn create_snapshot(&self, result_tx: oneshot::Sender<anyhow::Result<SnapshotId>>) {
+    pub fn can_create_snapshot(&self) -> bool {
```

Comment:
Maybe add Rust doc stating what's the condition for being able to create snapshots (I guess being fully caught up with the log?).